### PR TITLE
test: unit xmir test for rule 2 transformation

### LINKED ISSUE #50: 2-1-substitute-applications.xsl:29-32: add unit-test for...
The puzzle `47-c539e2d9` from #47 has to be resolved: 

https://github.com/objectionary/speco/blob/d3260f62cc2875b79dd473bcebce0213fa4d63cc/src/main/resources/org/eolang/speco/2-1-substitute-applications.xsl#L29-L32

The puzzle was created by @rultor on 15-Feb-23. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by advasileva**
Add unit test for `2-1-substitute-applications.xsl` transformation. Closes #50

### CODE REVIEW

**Comment by yegor256** [comment]
@volodya-lombrozo please help

File: src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          <o line="9" name="x" pos="1"/>
+          <o base=".boom" line="10" name="@" pos="3">
+            <o base="x" line="10" pos="2" ref="9"/>
+          </o>
+        </o>
+      </version>
+    </speco>
+  </program>
+asserts:
+  - /program/objects/o/o/o[1][@base="foo_spec_x_first"]
```
Comment (volodya-lombrozo) [inline src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml:42]: @advasileva Again, I'm not sure here, but maybe we can check that we don't have objects with `[@base=foo]` anymore?

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          <o base="foo" line="14" pos="4" ref="9">
+            <o base="first" line="14" pos="8" ref="1"/>
+          </o>
+          <o base="foo" line="15" pos="4" ref="9">
+            <o base="second" line="15" pos="8" ref="5"/>
+          </o>
+        </o>
+      </o>
+    </objects>
+    <aoi/>
```
[the remark this answers is not in the document]
Comment (advasileva) [reply src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml:21]: @volodya-lombrozo In reality, this section looks like this:

```xml
   <aoi>
      <obj fqn="foo.x">
         <inferred>
            <obj fqn="first"/>
            <obj fqn="second"/>
         </inferred>
      </obj>
   </aoi>
```

That is, speco leaves it in its original form, but I deleted the inner part to shorten the test. I can delete it completely, but it seems to me that this way the structure is cleaner

[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml:21]: @advasileva Do you use `aoi` section further?

File: src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml (hunk 1 above)
Comment (advasileva) [reply src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml:42]: @volodya-lombrozo Good idea, now I will add such an assert

File: src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (advasileva) [reply src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml:21]: @volodya-lombrozo No, I don't use it, but it seems to me correct to leave this section untouched by default. For deletion, it is supposed to create a separate command line option within #11.

[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml:21]: @advasileva Ok, I see, let's discuss it in the #11 then. Thanks.

File: src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml (hunk 1 above)
Comment (advasileva) [reply src/test/resources/org/eolang/speco/transformations/2-1-substitute-applications.yaml:42]: @volodya-lombrozo Done

**Review by volodya-lombrozo — APPROVED** [review]
@advasileva Looks good to me. Thanks.
@yegor256 Could you merge that changes, please?

**Comment by advasileva** [comment]
@yegor256 This one can also be merged)

**Comment by yegor256** [comment]
@rultor merge

**Comment by advasileva** [comment]
@yegor256 Thanks!

**Pull request merged**